syntax = "proto3";

package configuration;

service PartidaService {
    //Recebe o nome dos jogadores
    rpc CriarJogador(NomeJogador) returns (NomeJogadorResp) {};

    //Define a palavra que ser√° adivinhada na partida
    rpc EscolherPalavra(Palavra) returns (PalavraResp){};

    //O jogador que fornece a dica deve receber a palavra
    rpc VerPalavra(MensagemVazia) returns (Palavra) {};
}

message NomeJogador {
    string nome = 1;
}

message NomeJogadorResp {
    string nome = 1;
    bool recebida = 2;
}

message Palavra {
    string palavra = 1;
}

message PalavraResp {
    string palavra = 1;
    bool recebida = 2;
}

message MensagemVazia {

}